src/sources/mqtt/config.rs

```diff
@@ -0,0 +1,203 @@
+use std::time::Duration;
+
+use rand::Rng;
+use rumqttc::{MqttOptions, TlsConfiguration, Transport};
+use snafu::{ResultExt, Snafu};
+use vector_lib::codecs::decoding::{DeserializerConfig, FramingConfig};
+use vector_lib::config::{LegacyKey, LogNamespace};
+use vector_lib::configurable::configurable_component;
+use vector_lib::lookup::owned_value_path;
+use vector_lib::tls::{MaybeTlsSettings, TlsEnableableConfig};
```

Comment:
Done.

For the record nightly rustfmt can do this automatically, I think it's worth it to consider it. We've been using it for some time now and it seems stable though we don't really use latest features.

Reply:
Nice. We'd need to wait for that to move into stable since we have so many different contributors and we shouldn't  dictate the nightly versions. But I'm definitely looking forward to that.